### PR TITLE
Update the Activity page to show UOPS metrics

### PR DESCRIPTION
**Pull request by maciekzygmunt**
Resolves L2B-7792 L2B-7825 L2B-8098

### CODE REVIEW

File: packages/frontend/src/components/chart/activity/project-activity-chart.tsx [hunk 1]
```diff
@@ -75,8 +75,8 @@ export function ProjectActivityChart({ milestones, projectId }: Props) {
           >
             <div className="flex flex-row items-center gap-2">
               <EthereumLineIcon className="hidden h-1.5 w-2.5 sm:inline-block" />
-              <span className="max-lg:hidden">ETH Mainnet Transactions</span>
-              <span className="lg:hidden">ETH Txs</span>
+              <span className="max-lg:hidden">ETH Mainnet Operations</span>
```
Comment (torztomasz) [inline packages/frontend/src/components/chart/activity/project-activity-chart.tsx:78]: I guess it should be user operations

File: packages/frontend/src/components/chart/activity/activity-chart.tsx [hunk 2]
```diff
@@ -97,10 +97,8 @@ export function ActivityChart({ milestones, entries }: Props) {
             >
               <div className="flex flex-row items-center gap-2">
                 <EthereumLineIcon className="hidden h-1.5 w-2.5 sm:inline-block" />
-                <span className="hidden md:inline">
-                  ETH Mainnet Transactions
-                </span>
-                <span className="md:hidden">ETH Txs</span>
+                <span className="hidden md:inline">ETH Mainnet Operations</span>
```
Comment (torztomasz) [inline packages/frontend/src/components/chart/activity/activity-chart.tsx:100]: I guess it should be user operations

File: packages/frontend/src/utils/number-format/format-uops-ratio.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { formatNumber } from './format-number'
+
+export function formatUopsRatio(ratio: number): string {
+  switch (ratio < 1.01) {
+    case true:
+      return '<1.01'
+    case false:
+      return formatNumber(ratio, 2)
+  }
```
Comment (torztomasz) [inline packages/frontend/src/utils/number-format/format-uops-ratio.ts:9]: STOP PLS

Comment (torztomasz) [reply packages/frontend/src/utils/number-format/format-uops-ratio.ts:9]:
```suggestion
  if (ratio < 1.01) {
    return '<1.01'
  }
  
  return formatNumber(ratio, 2)
```

**Review by torztomasz — CHANGES_REQUESTED** [review]

File: packages/frontend/src/utils/number-format/format-uops-ratio.ts (hunk 3 above)
Comment (lucadonnoh) [reply packages/frontend/src/utils/number-format/format-uops-ratio.ts]: lmaoooooooo

**Review by torztomasz — APPROVED** [review]

File: packages/frontend/src/app/(side-nav)/scaling/activity/_components/table/columns.tsx [hunk 4]
```diff
@@ -71,10 +73,24 @@ export const getScalingActivityColumns = (
...
+        <SyncStatusWrapper syncStatus={data.syncStatus}>
+          <PrimaryValueCell>{formatUopsRatio(data.ratio)}</PrimaryValueCell>
+        </SyncStatusWrapper>
+      )
+    },
     meta: {
-      headClassName: 'w-0',
+      align: 'right',
+      tooltip:
+        'The ratio of user operations to transactions over the past day.',
```
Comment (radinat) [inline packages/frontend/src/app/(side-nav)/scaling/activity/_components/table/columns.tsx:93]:
```suggestion
        'The ratio of user operations to transactions over the past day. A high ratio indicates that most transactions contain multiple user operations bundled into a single transaction.' ,
```

**Review by radinat — CHANGES_REQUESTED** [review]

File: packages/frontend/src/app/(side-nav)/scaling/activity/_components/table/columns.tsx [hunk 5]
```diff
@@ -71,10 +73,24 @@ export const getScalingActivityColumns = (
...
+        <SyncStatusWrapper syncStatus={data.syncStatus}>
+          <PrimaryValueCell>{formatUopsRatio(data.ratio)}</PrimaryValueCell>
+        </SyncStatusWrapper>
+      )
+    },
     meta: {
-      headClassName: 'w-0',
+      align: 'right',
+      tooltip:
+        'The ratio of user operations to transactions over the past day. A high ratio indicates that most transactions contain multiple user operations bundled into a single transaction.',
```
Comment (radinat) [inline packages/frontend/src/app/(side-nav)/scaling/activity/_components/table/columns.tsx:93]:
```suggestion
        'The ratio of user operations to transactions over the past day. A high ratio indicates that for some transactions multiple individual user operations are bundled in a single transaction.',
```

**Review by radinat — CHANGES_REQUESTED** [review]

File: packages/frontend/src/app/(side-nav)/scaling/_components/scaling-activity-filters.tsx [hunk 6]
```diff
@@ -38,10 +40,27 @@ export function ScalingActivityFilters({ items, className }: Props) {
...
 
   return (
-    <ScalingFilters
-      items={items}
-      additionalFilters={daLayerFilter}
-      className={className}
-    />
+    <div
+      className={cn(
+        'flex flex-col-reverse gap-x-4 gap-y-2 [@media(min-width:1000px)]:flex-row [@media(min-width:1000px)]:justify-between',
```
Comment (torztomasz) [inline packages/frontend/src/app/(side-nav)/scaling/_components/scaling-activity-filters.tsx:45]: container queries mb?

File: packages/frontend/src/app/(side-nav)/scaling/activity/_components/table/max-uops-cell.tsx [hunk 7]
```diff
@@ -1,18 +1,18 @@
...
+import { formatUops } from '~/utils/number-format/format-uops'
 
 interface Props {
-  maxTps: number
+  maxUops: number
   timestamp: number
 }
 
-export function MaxTpsCell({ maxTps, timestamp }: Props) {
+export function MaxUopsCell({ maxUops, timestamp }: Props) {
```
Comment (torztomasz) [inline packages/frontend/src/app/(side-nav)/scaling/activity/_components/table/max-uops-cell.tsx:11]: its not only UOPS but also TPS, name should be somehow unified

File: packages/frontend/src/app/(side-nav)/scaling/summary/_components/table/columns.tsx [hunk 8]
```diff
@@ -90,24 +90,20 @@ export const scalingSummaryColumns = [
...
       const data = ctx.row.original.activity
-      if (!data) {
-        return <NoDataBadge />
-      }
-
       return (
-        <ValueWithPercentageChange change={data.change}>
-          {formatTps(ctx.getValue())}
+        <ValueWithPercentageChange change={data?.change}>
+          {formatUops(ctx.getValue())}
```
Comment (torztomasz) [inline packages/frontend/src/app/(side-nav)/scaling/summary/_components/table/columns.tsx:99]: this fn is probably also used with TPS so it should have universal name

**Review by torztomasz — CHANGES_REQUESTED** [review]

**Review by torztomasz — APPROVED** [review]

**Pull request merged**